packages/react/src/components/GroupedList/GroupedListV2.base.tsx

```diff
@@ -0,0 +1,604 @@
+    onRenderShowAll = renderGroupShowAll,
+  } = groupProps;
+
+  const classNames: IProcessedStyleSet<IGroupedListStyles> = getClassNames(styles, {
+    theme: theme!,
+    className,
+    compact,
+  });
+
+  const events = React.useRef<EventGroup>();
+  const flatList = React.useRef<IGroupedItem[]>([]);
+  const isSomeGroupExpanded = React.useRef<boolean>(computeIsSomeGroupExpanded(groups));
+
+  const [version, setVersion] = React.useState({});
+  const [toggleVersion, setToggleVersion] = React.useState({});
+
+  // eslint-disable-next-line deprecation/deprecation
+  const { shouldEnterInnerZone = isInnerZoneKeystroke } = focusZoneProps;
+
+  const listView = React.useMemo(() => {
+    return flattenItems(groups, items, flatList.current, groupProps);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [groups, groupProps, items, toggleVersion, flatList]);
+
+  const getPageSpecification = React.useCallback(
+    (flattenedIndex: number): { key?: string } => {
+      const pageGroup = listView[flattenedIndex];
+      return {
+        key: pageGroup.type === 'header' ? pageGroup.group.key : undefined,
+      };
+    },
+    [listView],
+  );
+
+  useMount(() => {
+    if (groupProps?.isAllGroupsCollapsed) {
+      setGroupsCollapsedState(groups, groupProps.isAllGroupsCollapsed);
+    }
+    events.current = new EventGroup(this);
+  });
+
+  useUnmount(() => {
+    events.current?.dispose();
+  });
+
+  React.useEffect(() => {
+    setVersion({});
+  }, [compact]);
+
+  React.useEffect(() => {
+    const newIsSomeGroupExpanded = computeIsSomeGroupExpanded(groups);
+    if (newIsSomeGroupExpanded !== isSomeGroupExpanded.current) {
+      isSomeGroupExpanded.current = newIsSomeGroupExpanded;
+      onGroupExpandStateChanged?.(newIsSomeGroupExpanded);
+    }
+  }, [groups, toggleVersion, onGroupExpandStateChanged, groupExpandedVersion]);
+
+  const onToggleCollapse = React.useCallback(
+    (group: IGroup): void => {
+      const onToggleCollapseFn = groupProps?.headerProps?.onToggleCollapse;
+
+      if (group) {
+        onToggleCollapseFn?.(group);
+        group.isCollapsed = !group.isCollapsed;
+        setToggleVersion({});
+        setVersion({});
+      }
+    },
+    [setToggleVersion, groupProps],
+  );
+
+  const onToggleSelectGroup = React.useCallback(
+    (group: IGroup): void => {
+      if (group && selection && selectionMode === SelectionMode.multiple) {
+        selection.toggleRangeSelected(group.startIndex, group.count);
+      }
+    },
+    [selection, selectionMode],
+  );
+
+  const onToggleSummarize = React.useCallback(
+    (group: IGroup): void => {
+      const onToggleSummarizeFn = groupProps?.showAllProps?.onToggleSummarize;
+
+      if (onToggleSummarizeFn) {
+        onToggleSummarizeFn(group);
+      } else {
+        if (group) {
+          group.isShowingAll = !group.isShowingAll;
+        }
+
+        setVersion({});
+        setToggleVersion({});
+      }
+    },
+    [groupProps],
```

Comment:
I removed most of the memo code and ran Flamegrill and profiled in DevTools.

TLDR is that it's mostly better to remove memoization though some things get marginally slower. In either case V2 is on par with V1.

I've decided to remove memoization for all the render functions as its a net win for performance. I've kept it for the `flattenItems` function because running that every render will be a performance drag, especially for larger and/or deeply nested trees.

### Flamegrill

Perf numbers range from 2-4x without memo for layout and style recalc but they were already so low that this doesn't really move the needle.

### DevTools

Scripting costs are significantly lower than with GroupedListV1 or the prior V2 implementation. Render and paint costs are slightly higher.

While not faster in all cases my takeaway is that removing memo makes the code generally more maintainable and is a net perf win.